src/Queue/CachedTransformer.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace eLife\Bus\Queue;
+
+use Doctrine\Common\Cache\Cache;
+use eLife\ApiSdk\ApiSdk;
+use eLife\ApiSdk\Model\Model;
+use Psr\Log\LoggerInterface;
+
+class CachedTransformer implements QueueItemTransformer
+{
+    use BasicTransformer;
+
+    private $cache;
+    private $logger;
+    private $lifetime;
+    private $sdk;
+    private $serializer;
+    private $shouldCache;
+
+    public function __construct(
+        ApiSdk $sdk,
+        Cache $cache,
+        LoggerInterface $logger,
+        int $lifetime,
+        callable $shouldCache = null
+    ) {
+        $this->serializer = $sdk->getSerializer();
+        $this->sdk = $sdk;
+        $this->cache = $cache;
+        $this->logger = $logger;
+        $this->lifetime = $lifetime;
+        $this->shouldCache = $shouldCache;
+    }
+
+    private function shouldCacheEntity(string $type, string $id)
+    {
+        if ($shouldCache = $this->shouldCache) {
+            return $shouldCache($type, $id);
+        }
+
+        return true;
+    }
+
+    public function get($id, $type)
```

Comment:
an interface for this `get()` method would be useful as code in the serving phase of the service will not depend on a `CachedTransformer` but rather on a smaller interface